docs/lib/introduction.md

```diff
@@ -0,0 +1,789 @@
+specific _RepositorySpec_ representation. Therefore, its constraint merely consists of the _Repository Type_.  
+As it is rather cumbersome to enter JSON objects such as a RepositorySpec through the command line and as there may be
+multiple types of Repository Specs for certain Repository Types (e.g. legacy types), the ocm-cli allows to enter a
+uniform string representation called `UniformRepositorySpec` of a Repository Spec. Be aware, there are again different
+UniformRepositorySpecs for [OCM Repositories](../../pkg/contexts/ocm/internal/uniform.go) and for
+[OCI Repositories](../../pkg/contexts/oci/internal/uniform.go). Furthermore, _ocm_ and _oci references_, or rather
+_reference strings_ (e.g. `eu.gcr.io/gardener-project/landscaper/examples/charts/hello-world:1.0.0`) are parsed into a
+[representation that is based on a `UniformRepositorySpec`](../../pkg/contexts/oci/ref.go). Thus, to get access to the
+respective Repository through the ocm-lib, this `UniformRepositorySpec` first has to be converted into a corresponding
+`RepositorySpec`.  
+In conclusion, a `RepositorySpecHandlers` _Constraint Registry_ is an object that maps a _repository type
+(= constraint)_ to a number of corresponding `RepositorySpecHandler` objects that might be able to convert a
+`UniformRepositorySpec` into the appropriate `RepositorySpec` for that _Repository Type_. Therefore, a
+`RepositorySpecHandler` has to implement the following [interface](../../pkg/contexts/ocm/internal/uniform.go):
+
+```
+type RepositorySpecHandler interface {
+   MapReference(ctx Context, u *UniformRepositorySpec) (RepositorySpec, error)
+}
+```
+
+A `RepositorySpecHandlers` Registry **is not** also a `HandlerRegistrationHandlerRegistry`.
+
+### Section 3.2.4 - Blob Handler Registry
+
+The `BlobHandlerRegistry` can be considered a _Constraint Registry_. To avoid confusions, let's first clarify that the
+name
+_Blob Handler_ might be misleading as the only purpose of a _Blob Handler_ is the upload of blobs. This is also
+immediately visible when looking at their [interface](../../pkg/contexts/ocm/internal/blobhandler.go):
+
+```
+type BlobHandler interface {
+   StoreBlob(blob BlobAccess, artType, hint string, global AccessSpec, ctx StorageContext) (AccessSpec, error)
+}
+```
+
+A much more suitable name would therefore be _Blob Uploader_.
+
+When a _Component Version_, or rather, a _Component Descriptor_, is imported into an _OCM Repository_, a decision has
+to be made about how to deal with the _artifacts_, or rather, the _artifact blobs_ (e.g. does it stay a local artifact
+stored alongside the _Component Descriptor_ or is it uploaded into an _OCI Registry_ as an independent _OCI Artifact_?).
+These decisions can be made within BlobHandlers. The `StoreBlob` method has to return an `AccessSpec` that can then be
+inserted into the corresponding new _Component Descriptor_.
+
+A `BlobHanderRegistry` **is** also a `HandlerRegistrationHandlerRegistry`.
```

Comment:
Same as above: If it is necessary to state this here so explicitly, then some reasoning would be nice.